File: helpers/b64helper.php
#!/usr/bin/env php
<?php
//  php b64helper.php encode < infile > outfile.b64
//  php b64helper.php decode < infile.b64 > outfile
$cmd = $argv[1] ?? '';
if ($cmd === 'encode') {
    $data = fopen('php://stdin','r');
    $buf = '';
    while (!feof($data)) $buf .= fgets($data, 8192);
    echo rtrim(base64_encode($buf), "\n");
} elseif ($cmd === 'decode') {
    $in = stream_get_contents(STDIN);
    $decoded = base64_decode($in, true);
    if ($decoded === false) { fwrite(STDERR, "Bad base64\n"); exit(2); }
    echo $decoded;
} else {
    fwrite(STDERR, "Usage: php b64helper.php encode|decode\n");
    exit(1);
}

File: helpers/b64helper.py
#!/usr/bin/env python3
# b64helper.py
import sys, base64
if len(sys.argv) < 2:
    sys.stderr.write("Usage: python3 b64helper.py encode|decode\n"); sys.exit(1)
mode = sys.argv[1]
data = sys.stdin.buffer.read()
if mode == 'encode':
    sys.stdout.write(base64.b64encode(data).decode('ascii'))
elif mode == 'decode':
    try:
        sys.stdout.buffer.write(base64.b64decode(data))
    except Exception:
        sys.stderr.write("Bad base64\n"); sys.exit(2)
else:
    sys.stderr.write("Usage: python3 b64helper.py encode|decode\n"); sys.exit(1)
